### PR TITLE
ococo

### PR DESCRIPTION
**Pull request by Takadonet**
Waiting on version 0.1.2.6 to be finished being built on bioconda for this tool be done.

### CODE REVIEW

File: tools/ococo/ococo.xml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+          <assert_contents>
+            <has_text text="AF=1.00;CS=2,0,0,0;COV=2"/>
+          </assert_contents>
+        </output>
+      </test>
+    </tests>
+    <help>
+      <![CDATA[
+
+Documentation available @ https://github.com/karel-brinda/ococo
```
Comment (bgruening) [inline tools/ococo/ococo.xml]: I would add here a little bit more documentation, for people isolated in a private network.

File: tools/ococo/ococo.xml [hunk 2]
```diff
@@ -0,0 +1,123 @@
+<tool id="ococo" name="ococo" version="0.1.2.6" >
+    <description>consensus caller on SAM/BAM</description>
+    <requirements>
+      <requirement type="package" version="0.1.2.6">ococo</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:"   level="fatal"   description="Unknown error" />
+    </stdio>
+    <command>
```
Comment (bgruening) [inline tools/ococo/ococo.xml]: Can you use `detect_errors` here and remove stdio.

File: tools/ococo/ococo.xml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    #if $save_pileup:
+      --pileup '$pileup'
+    #end if
+
+      ]]>
+    </command>
+
+
+    <inputs>
+      <param name="input_alignment" type="data" format="bam,sam" label="BAM dataset"/>
```
Comment (bgruening) [inline tools/ococo/ococo.xml]: Alignment file in BAM or SAM format?

File: tools/ococo/ococo.xml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      <param name="save_vcf" type="boolean" truevalue="" falsevalue="" checked="false" label="Save consensus VCF" />
+      <param name="save_fasta" type="boolean" truevalue="" falsevalue="" checked="false" label="Save consensus fasta" />
+      <param name="save_pileup" type="boolean" truevalue="" falsevalue="" checked="false" label="Save pileup fasta" />
+      
+      
+      <param name="min_mq" type="integer" label="Minimum mapping quality score" help="skip alignments with mapping quality smaller than INT [1]" value="1"/>
+      <param name="min_bq" type="integer" label="Minimum base quality score" help="skip bases with base quality smaller than INT [13]" value="13"/>
+      <param name="ref_weight" type="integer" label="Initial nucleotides counter" help="initial counter value for nucleotides from ref [0]" value="0"/>
+      <param name="min_cov" type="integer" label="Minimum coverage required" help="minimum coverage required for update [2]" value="2"/>
+      <param name="maj_thres" type="float" label="Majority consensus threshold [0.51]"  value="0.51"/>
```
Comment (bgruening) [inline tools/ococo/ococo.xml]: You could add `argument` as well if you like.

File: tools/ococo/ococo.xml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+        <option value="ococo32">ococo32 (7b/counter, 32b/position)</option>
+        <option value="ococo64">ococo64 (15b/counter, 64b/position)</option>
+      </param>
+      <param name="strategy" type="select" label="Strategy for updates: [majority]">
+        <option value="majority">Majority (update to majority base)</option>
+        <option value="stochastic">stochastic (update to stochastically drawn base)</option>
+      </param>
+
+
+      <param name="save_vcf" type="boolean" truevalue="" falsevalue="" checked="false" label="Save consensus VCF" />
```
Comment (bgruening) [inline tools/ococo/ococo.xml]: An alternative would be to have a multiple-select field here.

File: tools/ococo/.shed.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+categories:
+- Variant Analysis
+description: Variant detection of SNVs
+homepage_url: https://github.com/karel-brinda/ococo
+long_description:  Variant detection from high-throughput sequencing data is an essential step in identification of alleles involved in complex diseases and cancer. To deal with these massive data, elaborated sequence analysis pipelines are employed. A core component of such pipelines is a read mapping module whose accuracy strongly affects the quality of resulting variant calls.  We propose a dynamic read mapping approach that significantly improves read alignment accuracy. The general idea of dynamic mapping is to continuously update the reference sequence on the basis of previously computed read alignments. Even though this concept already appeared in the literature, we believe that our work provides the first comprehensive analysis of this approach. 
+name: ococo
+owner: iuc
+remote_repository_url: https://github.com/karel-brinda/ococo
```
Comment (bgruening) [inline tools/ococo/.shed.yml]: This needs to point to the Galaxy tool, e.g. the iuc-repo pointing to the ococo directory.

**Comment by bgruening** [comment]
Thanks @Takadonet, just a few minor comments inline.

**Comment by Takadonet** [comment]
Thanks. Will take look Monday. Just really wanted to push out onto github before weekend!

File: tools/ococo/ococo.xml (hunk 2 above)
Comment (Takadonet) [reply tools/ococo/ococo.xml]: Fixed.

File: tools/ococo/ococo.xml (hunk 3 above)
Comment (Takadonet) [reply tools/ococo/ococo.xml]: Fixed.

File: tools/ococo/ococo.xml (hunk 4 above)
Comment (Takadonet) [reply tools/ococo/ococo.xml]: Added.

File: tools/ococo/ococo.xml (hunk 5 above)
Comment (Takadonet) [reply tools/ococo/ococo.xml]: Addressed

File: tools/ococo/.shed.yml (hunk 6 above)
Comment (Takadonet) [reply tools/ococo/.shed.yml]: Changed.

File: tools/ococo/ococo.xml (hunk 1 above)
Comment (Takadonet) [reply tools/ococo/ococo.xml]: Added more docs.

**Comment by bgruening** [comment]
@Takadonet is this still WIP?

**Comment by Takadonet** [comment]
@bgruening Does it look good?

**Comment by bgruening** [comment]
@Takadonet does the last commit look good to you?

**Comment by Takadonet** [comment]
Sounds good.

**Comment by bgruening** [comment]
Thanks!

**Pull request merged**